### PR TITLE
Feature/custom regions

### PR DESCRIPTION
**Pull request by sorja**
Contains changes:
new table regions
removed from country table columns: `{list_name_,full_name}{en,fr,es,ru}`, region
changed country repository queries to join regions as new field `regions[]`

### CODE REVIEW

File: common/i18n/resources/en/area.js [hunk 1]
```diff
@@ -8,6 +8,8 @@ module.exports = {
   NA: { listName: 'North and Central America' },
   OC: { listName: 'Oceania' },
   SA: { listName: 'South America' },
+  EU27: { listName: 'EU-27' },
```
Comment (minotogna) [inline common/i18n/resources/en/area.js:11]: add this to fr, es, ru as well

File: server/db/migration/migrations/sqls/20200930084905-custom-regions-up.sql [hunk 2]
```diff
@@ -0,0 +1,56 @@
+
```
Comment (minotogna) [inline server/db/migration/migrations/sqls/20200930084905-custom-regions-up.sql:1]: place drop commands on top

File: server/db/migration/migrations/sqls/20200930084905-custom-regions-up.sql [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+        ('EU27', 'LUX'),
+        ('EU27', 'MLT'),
+        ('EU27', 'NLD'),
+        ('EU27', 'POL'),
+        ('EU27', 'PRT'),
+        ('EU27', 'ROU'),
+        ('EU27', 'SVK'),
+        ('EU27', 'SVN'),
+        ('EU27', 'ESP'),
+        ('EU27', 'SWE');
```
Comment (minotogna) [inline server/db/migration/migrations/sqls/20200930084905-custom-regions-up.sql:56]: should we insert also the current county_region ?

File: webapp/app/country/countryState.js [hunk 4]
```diff
@@ -10,6 +10,9 @@ const keys = {
   config: 'config',
   status: 'status',
   countries: 'countries',
+  regions: 'regions',
+  countryRegions: 'countryRegions',
+  regionCountries: 'regionCountries',
```
Comment (minotogna) [inline webapp/app/country/countryState.js:15]: what's the difference between countryRegions and regionCountries?

File: server/db/migration/migrations/sqls/20200930084905-custom-regions-up.sql (hunk 3 above)
Comment (sorja) [reply server/db/migration/migrations/sqls/20200930084905-custom-regions-up.sql:56]: There is line 24:
` INSERT INTO country_region SELECT region_iso, country_iso FROM country; `
Is this what you meant?

File: webapp/app/country/countryState.js (hunk 4 above)
Comment (sorja) [reply webapp/app/country/countryState.js:15]: countryRegions means in which regions a country belongs, for ex. Finland can be part of many different regions.
regionCountries is the other way around - what countries belong to a region, ex. EU contains FIN but also other countries

Comment (minotogna) [reply webapp/app/country/countryState.js:15]: but why not to fetch regions together with the countries? so in the UI we have only countries with a property regionCodes (array) ? + the regions themselves .
would make this things easier ?

File: server/country/countryRepository.js [hunk 5]
```diff
@@ -74,27 +59,35 @@ const getAllCountries = (role, schemaName = 'public') => {
...
+    WHERE NOT (message IN ($1))
+    GROUP BY country_iso
+)
+SELECT c.country_iso,
+       c.pan_european,
+       a.type,
+       a.status,
+       a.desk_study,
+       fa.last_edited,
+       json_agg(cr.region_code) AS regions
```
Comment (minotogna) [inline server/country/countryRepository.js:78]: I would call the field region_codes

File: common/country/country.js [hunk 6]
```diff
@@ -3,8 +3,6 @@ const R = require('ramda')
 const keys = {
   countryIso: 'countryIso',
   assessment: 'assessment',
-  fullName: 'fullName',
-  listName: 'listName',
   lastEdit: 'lastEdit',
   region: 'region',
   regionIso: 'regionIso',
```
Comment (minotogna) [inline common/country/country.js:8]: do we still regionIso prop ?

File: server/db/migration/migrations/sqls/20201013053954-drop-country-extra-columns-up.sql [hunk 7]
```diff
@@ -0,0 +1,15 @@
...
+
+alter table country drop column full_name_es;
+
+alter table country drop column list_name_fr;
+
+alter table country drop column full_name_fr;
+
+alter table country drop column list_name_ru;
+
+alter table country drop column full_name_ru;
```
Comment (minotogna) [inline server/db/migration/migrations/sqls/20201013053954-drop-country-extra-columns-up.sql:15]: should we drop also region_iso column ?

File: webapp/app/country/countryState.js (hunk 4 above)
Comment (minotogna) [reply webapp/app/country/countryState.js:15]: why these 3 properties? can't you filter client side instead fetching all these?

**Review by minotogna — CHANGES_REQUESTED** [review]
Also, please check if data export (from admin section and bulk download) includes regions.

File: server/db/migration/migrations/sqls/20201013053954-drop-country-extra-columns-up.sql (hunk 7 above)
Comment (sorja) [reply server/db/migration/migrations/sqls/20201013053954-drop-country-extra-columns-up.sql:15]: It is dropped in another migration 
server/db/migration/migrations/sqls/20201001104113-country-remove-region-region-iso-up.sql

File: webapp/app/country/countryState.js (hunk 4 above)
Comment (minotogna) [reply webapp/app/country/countryState.js:15]: please answer this

**Review by minotogna — CHANGES_REQUESTED** [review]

File: webapp/app/country/countryState.js (hunk 4 above)
Comment (sorja) [reply webapp/app/country/countryState.js:15]: This was fixed, missed few keys.
Removing

**Review by minotogna — APPROVED** [review]

**Pull request merged**